go/vt/wrangler/migrater.go

```diff
@@ -332,12 +341,6 @@ func (mi *migrater) validate(ctx context.Context, isWrite bool) error {
 			return mi.validateTableForWrite(ctx)
 		}
 	} else { // binlogdatapb.MigrationType_SHARDS
-		// Source and target shards must not match.
-		for sourceShard := range mi.sources {
```

Comment:
Might be worth keeping this check, since validate is always called after buildMigrater. This will serve as an invariant in case of future changes to buildMigrater.